src/main/java/com/google/devtools/build/lib/sandbox/SandboxStash.java

```diff
@@ -332,4 +480,145 @@ private static boolean isTestXmlGenerationOrCoverageSpawn(
   private static String getCurrentRunfilesDir(Map<String, String> environment) {
     return environment.get("TEST_WORKSPACE") + "/" + environment.get(TEST_SRCDIR);
   }
+
+  private static void listContentsRecursively(
+      Path root, Long timestamp, StashContents stashContents)
```

Comment:
Could you add a comment here that explains how `stashContents` is related to the actual on-disk contents of the sandbox when this function is called and when it returns?